client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamImpl.java

```diff
@@ -95,55 +98,60 @@ public long getOffset() {
     @Synchronized
     public ByteBuffer read(long timeout) throws EndOfSegmentException {
         log.trace("Read called at offset {}", offset);
-        fillBuffer();
-        while (buffer.dataAvailable() < WireCommands.TYPE_PLUS_LENGTH_SIZE) {
-            if (buffer.dataAvailable() == 0 && receivedEndOfSegment) {
-                throw new EndOfSegmentException();
-            }
-            if (outstandingRequest.await(timeout)) {
-                handleRequest();
-            } else {
-                return null;
-            }
-        }
         long originalOffset = offset;
         boolean success = false;
         try {
-            headerReadingBuffer.clear();
-            offset += buffer.read(headerReadingBuffer);
-            headerReadingBuffer.flip();
-            int type = headerReadingBuffer.getInt();
-            int length = headerReadingBuffer.getInt();
-            if (type != WireCommandType.EVENT.getCode()) {
-                throw new InvalidMessageException("Event was of wrong type: " + type);
-            }
-            if (length < 0 || length > WireCommands.MAX_WIRECOMMAND_SIZE) {
-                throw new InvalidMessageException("Event of invalid length: " + length);
-            }
-            ByteBuffer result = ByteBuffer.allocate(length);
-            offset += buffer.read(result);
-            while (result.hasRemaining()) {
-                handleRequest();
-                offset += buffer.read(result);
-            }
+            ByteBuffer result = readEventData(timeout);
             success = true;
-            result.flip();
             return result;
         } finally {
             if (!success) {
+                outstandingRequest = null;
                 offset = originalOffset;
                 buffer.clear();
             }
         }
     }
 
+    private ByteBuffer readEventData(long timeout) throws EndOfSegmentException {
```

Comment:
I didn't compare line by line here, but it feels like you are just moving code around for organization purposes, is that right?